test/common.test.ts

```diff
@@ -56,6 +56,24 @@ describe('common unit tests', () => {
       const loginUrl = await Common.resolveLoginUrl(undefined);
       expect(loginUrl).to.equal(SfdcUrl.PRODUCTION);
     });
+    it('should throw on lightning login URL', async () => {
+      sandbox.stub(SfdxProject.prototype, 'resolveProjectConfig').resolves({
+        packageDirectories: [
+          {
+            path: 'force-app',
+            default: true,
+          },
+        ],
+        sfdcLoginUrl: 'https://shanedevhub.lightning.force.com',
+        sourceApiVersion: '50.0',
+      });
+      try {
+        await Common.resolveLoginUrl(undefined);
```

Comment:
Will this test work correctly if `Common.resolveLoginUrl` never throws?
I usually do something like put an `expect(1).to.equal('This test is failing because it is expecting an error that is never thrown')` after the function call to ensure it fails if the error handling logic around this scenario changes unexpectedly.

Reply:
You can also just use `assert.fail('This test is failing because it is expecting an error that is never thrown')`.